src/Common/tests/System.Net/Sockets/Performance/SocketTestClientAPM.cs

```diff
@@ -51,12 +60,23 @@ public override void Receive(Action<int> onReceiveCallback)
 
         private void OnReceive(IAsyncResult result)
         {
-            Action<int> callback = (Action<int>)result.AsyncState;
-            int recvBytes = _s.EndReceive(result);
-            callback(recvBytes);
+            Action<int, SocketError> callback = (Action<int, SocketError>)result.AsyncState;
```

Comment:
Do we need to be concerned about code that receives one of these IAsyncResults doing a similar cast and then invoking the callback?  Are these IAsyncResults exposed out to other code?  Or does the callback protect against that kind of misuse?  Not something to hold up this PR, just something to think through.